### PR TITLE
feat(ilp): breaking change💥 - ILP and TextImport create VARCHAR instead of STRING columns

### PR DESCRIPTION
**Pull request by jerrinot**
Text import and ILP now create VARCHAR columns instead of STRING columns. VARCHAR type uses less disk space and is faster than STRING for most operations. Most users should use VARCHAR columns instead of STRING columns.

If VARCHAR type is not yet usable for a given use case, then you can opt out by setting this configuration option: `cairo.legacy.string.column.type.default=true`.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 50 / 51 (98.04%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/text/types/VarcharAdapter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TypeManager.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CsvFileIndexer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyJob.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 55 / 57 (96.49%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/VarcharAdapter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/DefaultTextConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TypeManager.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CsvFileIndexer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyJob.java|5|5|100.00%|

**Review by ideoma — APPROVED** [review]

**Review by ideoma — DISMISSED** [review]

File: core/src/main/java/io/questdb/PropertyKey.java [hunk 1]
```diff
@@ -341,9 +341,9 @@ public enum PropertyKey implements ConfigPropertyKey {
     LINE_TCP_UNDOCUMENTED_STRING_AS_TAG_SUPPORTED("line.tcp.undocumented.string.as.tag.supported"),
     LINE_FLOAT_DEFAULT_COLUMN_TYPE("line.float.default.column.type"),
     LINE_INTEGER_DEFAULT_COLUMN_TYPE("line.integer.default.column.type"),
-    LINE_USE_LEGACY_STRING_DEFAULT("line.use.legacy.string.default"),
     LINE_TCP_NET_IO_QUEUE_CAPACITY("line.tcp.net.io.queue.capacity"),
     LINE_TCP_IO_AGGRESSIVE_RECV("line.tcp.io.aggressive.recv"),
+    USE_LEGACY_STRING_DEFAULT("use.legacy.string.default"),
```
[the remark this answers is not in the document]
Comment (jerrinot) [reply core/src/main/java/io/questdb/PropertyKey.java:346]: good one, thanks!

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 55 / 57 (96.49%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/VarcharAdapter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/DefaultTextConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TypeManager.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CsvFileIndexer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyJob.java|5|5|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 55 / 57 (96.49%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/VarcharAdapter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/DefaultTextConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TypeManager.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CsvFileIndexer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyJob.java|5|5|100.00%|

**Review by ideoma — APPROVED** [review]

**Comment by jerrinot** [comment]
@ideoma I resolved a merge conflict and it dismissed your review. please re-approve. thanks!

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 55 / 57 (96.49%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/VarcharAdapter.java|16|17|94.12%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TimestampUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/DefaultTextConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextMetadataDetector.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/DateUtf8Adapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/TypeManager.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CsvFileIndexer.java|5|5|100.00%|
|:large_blue_circle:|io/questdb/PropertyKey.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/ParallelCsvFileImporter.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/StringAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/types/SymbolAdapter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyTask.java|4|4|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/TextLoader.java|8|8|100.00%|
|:large_blue_circle:|io/questdb/cutlass/text/CopyJob.java|5|5|100.00%|

**Pull request merged**